implementations/rust/ockam/ockam_entity/src/credential/traits.rs

```diff
@@ -89,3 +100,151 @@ pub trait CredentialVerifier {
         proof_request_id: ProofRequestId,
     ) -> Result<bool>;
 }
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize, Default)]
+pub struct CredentialIdentifier(String);
+
+impl Distribution<CredentialIdentifier> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> CredentialIdentifier {
+        let id: [u8; 16] = rng.gen();
+        CredentialIdentifier(hex::encode(id))
+    }
+}
+
+impl CredentialIdentifier {
+    pub const PREFIX: &'static str = "C";
+}
+
+impl Display for CredentialIdentifier {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let str: String = self.clone().into();
+        write!(f, "{}", &str)
+    }
+}
+
+impl Into<String> for CredentialIdentifier {
+    fn into(self) -> String {
+        format!("{}{}", Self::PREFIX, &self.0)
+    }
+}
+
+impl TryFrom<&str> for CredentialIdentifier {
+    type Error = Error;
+
+    fn try_from(value: &str) -> Result<Self> {
+        if let Some(str) = value.strip_prefix(Self::PREFIX) {
+            Ok(Self(str.into()))
+        } else {
+            Err(EntityError::InvalidProfileId.into())
+        }
+    }
+}
+
+impl TryFrom<String> for CredentialIdentifier {
+    type Error = Error;
+
+    fn try_from(value: String) -> Result<Self> {
+        Self::try_from(value.as_str())
+    }
+}
+
+#[derive(Clone, Eq, PartialEq, Debug, Serialize, Deserialize)]
+pub struct Credential {
+    id: CredentialIdentifier,
+    issuer_id: EntityIdentifier,
+    type_id: String,
```

Comment:
Not much information is conveyed in the name `type_id` - this is really the identifier of the schema yeah?

Reply:
I think we want to stop exposing word "schema"... TypedId is what we ended up in the last meeting, suggestions are welcome of course